src/FragmentService/AbstractFragmentService.php

```diff
@@ -95,4 +107,16 @@ public function getEditTemplate(): string
     {
         return '@SonataArticle/FragmentAdmin/form.html.twig';
     }
+
+    public function setTranslator(TranslatorInterface $translator): void
+    {
+        $this->translator = $translator;
+    }
+
+    public function trans(string $id, array $parameters = [], string $domain = null, string $locale = null): string
+    {
+        $domain = ($domain) ? $domain : $this->translationDomain;
+
+        return $this->translator->trans($id, $parameters, $domain, $locale);
+    }
```

Comment:
Why would translation be the responsability of a fragment? This class is already quite long, I'm relunctant to cramming more things in it, especially because does not use existing member variables. It could live somewhere else but just sits there in the class, not sharing anything with other methods… also, it will crash if `setTranslator` is not called.

Reply:
I agree with you, i m just proposing a solution  ^^ You have an alternative how to translate the fragment name otherwise ?

Reply:
a specific Twig filter, named `fragment_trans`?

Reply:
Done ^^